unity/Holo/Assets/Editor/ModelPreprocessor.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+class ModelPreprocessor
+{
+    public string rootDirectory;
+
+    private GameObject modelGameObject;
+    private Mesh mesh;
+    private AssetBundleCreator assetBundleCreator = new AssetBundleCreator();
+    private List<string> assetFilePaths = new List<string>();
+
+    /* To use in batchmode: "<Path to Unity.exe>" -quit -batchmode -logFile "<Path to the logfile>" 
+     -executeMethod ModelPreprocessor.PreprocessModel -rootDirectory "<Directory of the folder which stores the meshes>"
+    */
+    [MenuItem("Holo/Convert model to a AssetBundle's GameObject")]
+    public static void PreprocessModel()
+    {
+        ModelPreprocessor modelPreprocessor = new ModelPreprocessor();
+
+        modelPreprocessor.GetRootDirectory();
+        modelPreprocessor.GetModelData();
+        modelPreprocessor.assetBundleCreator.Create(modelPreprocessor.modelGameObject, modelPreprocessor.mesh);
+    }
+
+    // gets path to the root folder.
+    private void GetRootDirectory()
+    {
+        if (Application.isBatchMode)
+        {
+            Debug.Log("It's Batchmode!");
+            string[] args = Environment.GetCommandLineArgs();
+            bool rootDirExists = false;
+            for (int i = 0; i < args.Length; i++)

```

Comment:
It would be much easier and cleaner with Linq
``` string rootDirectory = args.Single( a => a == "-rootDirectory");```
```if(String.IsNullOrEmpty(rootDirectory)) throw ...```

Reply:
As "-rootDirectory" is a separate element, wouldn't it mean rootDirectory = "-rootDirectory" instead of an actual directory?